src/DurableTask.AzureStorage/Tracking/AzureTableTrackingStore.cs

```diff
@@ -953,7 +957,15 @@ public override Task StartAsync()
                     ["LastUpdatedTime"] = new EntityProperty(newEvents.Last().Timestamp),
                 }
             };
-           
+
+            // check if we are replacing a previous execution with blobs; those will be deleted from the store after the update
+            List<string> blobsToDelete = null;
+            if (oldRuntimeState != newRuntimeState && data.Blobs.Count > 0)
+            {
+                blobsToDelete = data.Blobs;
+                data.Blobs = new List<string>();
+            }
```

Comment:
nit: let's call out explicitly this is a ContinueAsNew scenario. I think it's helpful to convey the high-level API that would drive this.
```suggestion
            // check if we are replacing a previous execution with blobs; those will be deleted from the store after the update. This could occur in a ContinueAsNew scenario
            List<string> blobsToDelete = null;
            if (oldRuntimeState != newRuntimeState && data.Blobs.Count > 0)
            {
                blobsToDelete = data.Blobs;
                data.Blobs = new List<string>();
            }
```